todo-app/src/components/todoList.js

```diff
@@ -0,0 +1,45 @@
+import React, { Component } from "react";
+import TaskItem from "./taskItem.js";
+import AddingContainer from "./addingContainer.js"
+
+export let todoArray = [
+    { description: 'Get out of bed', deadLine: 'Wed Sep 13 2017' },
+    { description: 'Brush teeth', deadLine: 'Thu Sep 14 2017' },
+    { description: 'Eat breakfast', deadLine: 'Fri Sep 15 2017' }
+]
+
+class TodoList extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            tasks: todoArray
+        }
+    }
+   
+    addingTask = (description, deadLine) => {
+        this.setState((previousState) => ({
+                tasks: [ { description: description, deadLine: deadLine }, ...previousState.tasks]
+            })
+        )
+        console.log(this.state)
```

Comment:
It doesn't make sense to try and `console.log` out state below your `setState` call, as `setState` is an _asynchronous_ request that does not necessarily execute immediately. A small distinction, but very important to keep in mind.

You should think of `setState` as a future request made to React to update our UI, not an immediate change being made. React will make the requested changes at the next possible moment in terms of performance.